vllm/config.py

```diff
@@ -3918,12 +3918,14 @@ class CompilationConfig:
     constructor, e.g. `CompilationConfig(inductor_passes={"a": func})`."""
 
     # CudaGraph compilation
-    use_cudagraph: bool = False
+    use_cudagraph: bool = envs.VLLM_USE_V1
```

Comment:
I don't think this is the right default - this should probably depend on the compilation level.

Reply:
False was the right default for V0.

For V1, we just override all compilation levels to be PIECEWISE, so True is the right default for VLLM_USE_V1. We force override -O1 and -O2 to be -O3. I'll fix this in a separate PR, but this PR should have no behavior changes.

Reply:
Wondering if we should give it a better name like piece_wise_use_cudagraph or not.

Reply:
I think this name is fine because this flag also controls full cudagraphs. But perhaps we can consolidate them